patch_files/patches/46_resume_cancelled_cloudify_workflows_workflow_context.py

```diff
@@ -0,0 +1,37 @@
+diff --git a/cloudify/workflows/workflow_context.py b/cloudify/workflows/workflow_context.py
+index af9adf9..6a05d18 100644
+--- a/cloudify/workflows/workflow_context.py
++++ b/cloudify/workflows/workflow_context.py
+@@ -1270,14 +1270,13 @@ class _TaskDispatcher(object):
+         ping_handler = amqp_client.BlockingRequestResponseHandler(
+             exchange=task['target'])
+         client = self._get_client(task)
+-        client.add_handler(handler)
+         client.add_handler(ping_handler)
++        client.add_handler(handler)
+         task.update({
+             'client': client,
+             'handler': handler,
+             'ping_handler': ping_handler
+         })
+-        client.consume_in_thread()
+         return task
+ 
+     def _get_client(self, task):
+@@ -1318,6 +1317,7 @@ class _TaskDispatcher(object):
+ 
+         self._tasks.setdefault(client, {})[task['id']] = \
+             (workflow_task, task, result)
++        client.consume_in_thread()
+         return result
+ 
+     def _set_task_state(self, workflow_task, state, event=None):
+@@ -1328,7 +1328,7 @@ class _TaskDispatcher(object):
+                     state, workflow_task,
+                     events.send_task_event_func_remote, event)
+ 
+-    def _received(self, task_id, client, response):
++    def _received(self, task_id, client, response, channel):
```

Comment:
this change isn't needed and will be reverted (it's the other side of the to-be-reverted change in amqp_client.py)